internal/server/service/secretService.go

```diff
@@ -0,0 +1,79 @@
+// Package service предоставляет бизнес-логику для управления секретами пользователей.
+package service
+
+import (
+	"beliaev-aa/GophKeeper/internal/server/storage/repository"
+	"beliaev-aa/GophKeeper/pkg/models"
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+)
+
+// SecretService предоставляет методы для управления секретами в хранилище.
+type SecretService struct {
+	secretRepository *repository.SecretRepository // secretRepository является репозиторием для доступа к секретам в базе данных.
+}
+
+// NewSecretService создает новый экземпляр SecretService.
+// Принимает в качестве аргумента репозиторий секретов и возвращает ссылку на сервис.
+func NewSecretService(secretRepository *repository.SecretRepository) *SecretService {
+	return &SecretService{secretRepository: secretRepository}
+}
+
+// GetSecret извлекает секрет по его ID и ID пользователя.
+// В случае отсутствия секрета возвращает ошибку.
+func (s SecretService) GetSecret(ctx context.Context, secretID uint64, userID uint64) (*models.Secret, error) {
```

Comment:
Я бы рекомендовал в общем случае использовать pointer receiver.
```suggestion
func (s *SecretService) GetSecret(ctx context.Context, secretID uint64, userID uint64) (*models.Secret, error) {
```

Reply:
Исправил